### DISCUSSION TITLE
How to recieved SignalR Server Messeage from WinUI 3.

### DISCUSSION THREAD

**Discussion by KyawHtetSS** [microsoft/WindowsAppSDK#3561] [Q&A]
I'm testing SignalR and WinUI3 to add feature in existed application.

SignalR               7.0.4
SignalR Client     7.0.4
WinAppSDK        1.2.230313.1
Windows             22H2
Visual Studio        17.4.5.
.NET                    6 (both server and client)

I used following code in WinUI3 Project as Client.

    public sealed partial class MainWindow : Window
    {
        private HubConnection _hubConnection;

        public MainWindow()
        {
            this.InitializeComponent();

        }

        private async void Connect_Click(object sender, RoutedEventArgs e)
        {
            try
            {
                _hubConnection = new HubConnectionBuilder()
                .WithUrl(URLTextBox.Text.Trim())
                .Build();

                _hubConnection.Closed += async (error) =>
                {
                    await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, async () =>
                    {
                        await _hubConnection.StartAsync();
                    });
                };

                _hubConnection.On<string>("ReceiveMessage", async (message) =>
                {
                    MessagesTextBox.Text += $"{message}\r\n";
                    await Task.CompletedTask;
                });

                await _hubConnection.StartAsync();
                MessagesTextBox.Text += "Connected to server\r\n";
            }
            catch (Exception ex)
            {
                MessagesTextBox.Text += $"Error connecting to server: {ex.Message}\r\n";
            }
        }

        private async void Disconnect_Click(object sender, RoutedEventArgs e)
        {
            try
            {
                await _hubConnection.StopAsync();
                MessagesTextBox.Text += "Disconnected from server\r\n";
            }
            catch (Exception ex)
            {
                MessagesTextBox.Text += $"Error disconnecting from server: {ex.Message}\r\n";
            }
        }

        private async void Send_Click(object sender, RoutedEventArgs e)
        {
            try
            {
                await _hubConnection.InvokeAsync("SendMessage", "Client", MessageTextBox.Text);
                MessageTextBox.Text = "";
            }
            catch (Exception ex)
            {
                MessagesTextBox.Text += $"Error sending message: {ex.Message}\r\n";
            }
        }
    }

Following code in Server.

 

> public class NotiHub : Hub
>  {
>         public async Task SendMessage(string user, string message)
>         {
>             await Clients.All.SendAsync("ReceiveMessage", $"{user} : {message}");
> 
>             Console.WriteLine($"{user} : {message}");
>             //await Clients.All().SendAsync("ReceiveMessage", user, message);
>         }
>  }

>  public static void Main(string[] args)
>  {
>             var builder = WebApplication.CreateBuilder(args);
> 
>             // Add services to the container.
> 
>             builder.Services.AddControllers();
>             builder.Services.AddSignalR();
> 
>             var app = builder.Build();
> 
>             // Configure the HTTP request pipeline.
> 
>             app.UseHttpsRedirection();
> 
>             app.UseAuthorization();
> 
> 
>             app.MapControllers();
>             app.MapHub<NotiHub>("/NotiHub");
> 
> 
>             app.Run();
>  }

1. I can connect to SignalR server and Successfully connected. 
2. Connection ID is created. 
3. I can send message to SignalR Server.  
4. Server can broadcast message. 
5. But I cannot receive message from server in WinUI3 Client.
6. I tested with WinFormApp with .NET 6. It can successfully recieved with same code.

I've searched in web for sample or anything. There are none about SignalR with WinUI3. Is there anything I missed to do in WinUI3?

**Comment by koenvd** [comment]
Wild guess but enabling `Internet (Client and Server)` capabilities in your manifest file?
